internal/controller/filesystemclaim_controller.go

```diff
@@ -198,16 +183,34 @@ func (r *FileSystemClaimReconciler) handleFileSystemClaimRequest(ctx context.Con
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	// Add initial overall status condition
+	fsc.Status.Conditions = utils.UpdateCondition(fsc.Status.Conditions, ConditionTypeReady, metav1.ConditionFalse, ReasonProvisioningInProgress, "Provisioning in progress", fsc.Generation)
+	if statusErr := r.Status().Update(ctx, fsc); statusErr != nil {
```

Comment:
And doing this is worse.

Only update if something changed, and then exit from the reconcile loop (even if the update succeeds).
We make one change per pass through the loop.